### PR TITLE
Separate /ingest/icu into /tensorize/bedmaster and /tensorize/edw

### LINKED ISSUE #1036: Tensorize Bedmaster data without requiring EDW data or alarms
## What and why
Tensorizing only Bedmaster data still requires setting
```
--path_edw /nothing
--path_alarms /nothing
```

## Solution(s)
Pipeline should set defaults of `None` for those args if no paths are given, and skip over parsing those data when looking for data to tensorize.

### LINKED ISSUE #1098: Remove EDW from Bedmaster tensorization
## What and why
Cleaner organization of data and simpler code.

## Solution(s)
Refactor and simplify EDW tensorization pipeline.
Remove dependency on EDW information from Bedmaster tensorization pipeline. 

## Acceptance criteria
- Distinct `tensorize_bedmaster` recipes.
- Bedmaster data can be tensorized into distinct directory (see Discussion https://github.com/aguirre-lab/ml4c3/discussions/1095):
    ```
    /storage/shared
        /ecg             # hd5
        /edw             # hd5
        /bedmaster       # hd5
        /sts             # CSV file with tabular features
        /echo            # CSV file with tabular features
        /c3po            # CSV file with tabular features
        /aortic-stenosis # CSV file with tabular features
    ```

## Blocked by or pending

### PR DESCRIPTION
**Pull request by raimonpv**
## Issues closed
Closes #1036
Closes #1098 

## Changelog
- [x] This PR separates the content from `/ingest/icu` to two different directories: `/tensorize/bedmaster` and `/tensorize/edw`
TO DO:
- [ ] Modify tests accordingly to rely just in edw or bedmaster tensorization #1181 
- [ ] Modify `/ingest/icu/assess_coverage.py` and `pre_tensorize_explorations.py` to preserve functionality. The rest of the code is already separated. #1177  and #1178 
- [ ] Add alarms to tensorization #1180 
- [ ] General cleanup (there is still shared code between `/tensorize/bedmaster` and `/tensorize/edw`) #993 
- [ ] Once HD5 files with both edw and bedmaster data are not useful/needed, delete `/ingest/icu`

I think that we can open issues to implement the rest of the points.

## Checklist
- [ ] The following are complete:
    - Added or revised tests if appropriate.
    - If new packages are installed, `requirements.txt` is updated.
    - If documentation should be updated, a new issue was made in `ml4c3-wiki`.
    - The PR name describes its content and does not contain "Ref #".
    - The PR does not add PHI.

### CODE REVIEW

**Comment by erikr** [comment]
Since there are several other tasks you still want to do in this PR, want to convert it to a draft?

File: tensorize/edw/tensorizer.py [hunk 1]
```diff
@@ -0,0 +1,378 @@
...
+
+# Imports: first party
+from tensorize.utils import save_mrns_and_csns_csv
+from tensorize.edw.readers import EDWReader, CrossReferencer
+from tensorize.edw.writers import Writer
+
+
+class Tensorizer:
+    """
+    Main class to tensorize the Bedmaster data and the EDW data.
```
Comment (RogerPallares) [inline tensorize/edw/tensorizer.py:22]: Only EDW.

File: tensorize/bedmaster/tensorizer.py [hunk 2]
```diff
@@ -0,0 +1,408 @@
...
+from definitions.icu import ICU_SCALE_UNITS
+from tensorize.utils import save_mrns_and_csns_csv
+from tensorize.bedmaster.readers import BedmasterReader, CrossReferencer
+from tensorize.bedmaster.writers import Writer
+from tensorize.bedmaster.match_patient_bedmaster import PatientBedmasterMatcher
+
+
+class Tensorizer:
+    """
+    Main class to tensorize the Bedmaster data and the EDW data.
```
Comment (RogerPallares) [inline tensorize/bedmaster/tensorizer.py:24]: Only Bedmaster.

File: tensorize/bedmaster/readers.py [hunk 3]
```diff
@@ -0,0 +1,815 @@
...
+from definitions.edw import EDW_FILES
+from definitions.icu import ICU_SCALE_UNITS
+from tensorize.bedmaster.data_objects import BedmasterSignal
+from tensorize.bedmaster.bedmaster_stats import BedmasterStats
+from tensorize.bedmaster.match_patient_bedmaster import PatientBedmasterMatcher
+
+# pylint: disable=too-many-branches, dangerous-default-value
+
+
+class Reader(ABC):
```
Comment (RogerPallares) [inline tensorize/bedmaster/readers.py:25]: Should we have this class in `./tensorize` and share it with both Bedmaster and EDW?

File: tensorize/bedmaster/data_objects.py [hunk 4]
```diff
@@ -0,0 +1,155 @@
+# Imports: standard library
+from abc import ABC, abstractmethod
+
+# Imports: third party
+import numpy as np
+
+# pylint: disable=abstract-method, unnecessary-pass
+
+
+class ICUDataObject(ABC):
```
Comment (RogerPallares) [inline tensorize/bedmaster/data_objects.py:10]: Should we have this class in `./tensorize` and share it with both Bedmaster and EDW?

File: tensorize/edw/readers.py [hunk 5]
```diff
@@ -0,0 +1,831 @@
...
+    Procedure,
+    Medication,
+    StaticData,
+    Measurement,
+)
+
+# pylint: disable=too-many-branches, dangerous-default-value
+
+
+class Reader(ABC):
```
Comment (RogerPallares) [inline tensorize/edw/readers.py:28]: Should we have this class in `./tensorize` and share it with both Bedmaster and EDW?

File: tensorize/edw/data_objects.py [hunk 6]
```diff
@@ -0,0 +1,352 @@
...
+# Imports: third party
+import numpy as np
+
+# Imports: first party
+from definitions.edw import EDW_FILES
+
+# pylint: disable=abstract-method, unnecessary-pass
+
+
+class ICUDataObject(ABC):
```
Comment (RogerPallares) [inline tensorize/edw/data_objects.py:14]: Should we have this class in `./tensorize` and share it with both Bedmaster and EDW?

**Review by RogerPallares — CHANGES_REQUESTED** [review]
Great! Just some small changes.

File: tensorize/bedmaster/readers.py (hunk 3 above)
Comment (raimonpv) [reply tensorize/bedmaster/readers.py:25]: Definitely, there is an issue to make some cleanup as there is some shared code between both tensorizations #993

**Comment by StevenSong** [comment]
@erikr we ought to move ecg tensorization too

also is the idea behind the separation that ingest pulls raw data and tensorization formats

**Review by RogerPallares — APPROVED** [review]

**Pull request merged**